src/pybind/mgr/dashboard/frontend/src/app/core/auth/user-form/user-form.component.ts

```diff
@@ -59,39 +60,40 @@ export class UserFormComponent implements OnInit {
     private notificationService: NotificationService,
     private i18n: I18n,
     public actionLabels: ActionLabelsI18n,
-    private userChangePasswordService: UserChangePasswordService
+    private passwordPolicyService: PasswordPolicyService,
+    private formBuilder: CdFormBuilder
   ) {
     this.resource = this.i18n('user');
     this.createForm();
     this.messages = new SelectMessages({ empty: this.i18n('There are no roles.') }, this.i18n);
   }
 
   createForm() {
-    this.requiredPasswordRulesMessage = this.userChangePasswordService.getPasswordRulesMessage();
-    this.userForm = new CdFormGroup(
+    this.passwordPolicyHelpText = this.passwordPolicyService.getHelpText();
+    this.userForm = this.formBuilder.group(
```

Comment:
Any reason for this change? This will remove access to methods available in `CdFormGroup`.

Reply:
> Any reason for this change?

I was not able to get async validators running using the old code.

> This will remove access to methods available in CdFormGroup.

Yes correct, but they are not used here.

Reply:
Didn't notice `formBuilder` was using `CdFormBuilder `.